src/databricks/labs/remorph/reconcile/query_builder/threshold_query.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from sqlglot import expressions as exp
+from sqlglot import select
+
+from databricks.labs.remorph.reconcile.constants import ThresholdMode
+from databricks.labs.remorph.reconcile.query_builder.base import QueryBuilder
+from databricks.labs.remorph.reconcile.query_builder.expression_generator import (
+    anonymous,
+    build_column,
+    build_from_clause,
+    build_join_clause,
+    build_sub,
+    build_threshold_absolute_case,
+    build_threshold_percentage_case,
+    build_where_clause,
+    coalesce,
+)
+from databricks.labs.remorph.reconcile.recon_config import Thresholds
+from databricks.labs.remorph.snow.databricks import Databricks
+
+logger = logging.getLogger(__name__)
+
+
+class ThresholdQueryBuilder(QueryBuilder):
+    # Comparison query
+    def build_comparison_query(self) -> str:
+        select_clause, where = self._generate_select_where_clause()
+        from_clause, join_clause = self._generate_from_and_join_clause()
+        # for threshold comparison query the dialect is always Daabricks
+        query = select(*select_clause).from_(from_clause).join(join_clause).where(where).sql(dialect=Databricks)
+        logger.info(f"Threshold Comparison query: {query}")
+        return query
+
+    def _generate_select_where_clause(self) -> tuple[list[exp.Alias], exp.Or]:
+        thresholds = self.table_conf.thresholds
+
+        def _build_alias(this: str, table_name: str):
```

Comment:
avoid nested functions.

Reply:
Since there are already existing functions I thought it would be agreeable. Removed it